### PR TITLE
Update Password_Storage_Cheat_Sheet.md

### PR DESCRIPTION
**Pull request by jmanico**
Big update from Robert Thorton

Thank you for submitting a Pull Request (PR) to the Cheat Sheet Series. 

> :triangular_flag_on_post: If your PR is related to grammar/typo mistakes, please double-check the file for other mistakes in order to fix all the issues in the current cheat sheet.

Please make sure that for your contribution:

- [ ] In case of a new Cheat Sheet, you have used the [Cheat Sheet template](https://github.com/OWASP/CheatSheetSeries/blob/master/templates/New_CheatSheet.md).
- [ ] All the markdown files do not raise any validation policy violation, see the [policy](https://github.com/OWASP/CheatSheetSeries/actions?query=workflow%3A%22Markdown+Link+Check%22).
- [ ] All the markdown files follow these [format rules](https://github.com/OWASP/CheatSheetSeries/blob/master/CONTRIBUTING.md#markdown).
- [ ] All your assets are stored in the **assets** folder.
- [ ] All the images used are in the **PNG** format.
- [ ] Any references to websites have been formatted as [TEXT](URL)
- [ ] You verified/tested the effectiveness of your contribution (e.g., the defensive code proposed is really an effective remediation? Please verify it works!).
- [ ] The CI build of your PR pass, see the build status [here](https://github.com/OWASP/CheatSheetSeries/actions).

If your PR is related to an issue, please finish your PR text with the following line:

This PR covers issue #<insert number here>.

Thank you again for your contribution :smiley:

### CODE REVIEW

File: cheatsheets/Password_Storage_Cheat_Sheet.md [hunk 1]
```diff
@@ -153,24 +151,24 @@ The work factor for PBKDF2 is implemented through an iteration count, which shou
...
 
 ### International Characters
 
-Ensure your hashing library is able to accept a wide range of characters and is compatible with all Unicode codepoints. Users should be able to use the full range of characters available on modern devices, in particular mobile keyboards. They should be able to select passwords from various languages and include pictograms. Prior to hashing the entropy of the user's entry should not be reduced. Password hashing libraries need to be able to use input that may contain a NULL byte.
+Your hashing library must be able to accept a wide range of characters and should be compatible with all Unicode codepoints, so users can use the full range of characters available on modern devices--especially mobile keyboards. They should be able to select passwords from various languages and include pictograms. Prior to hashing the entropy of the user's entry should not be reduced, and password hashing libraries need to be able to use input that may contain a NULL byte.
```
Comment (szh) [inline cheatsheets/Password_Storage_Cheat_Sheet.md:174]:
```suggestion
Your hashing library must be able to accept a wide range of characters and should be compatible with all Unicode codepoints, so users can use the full range of characters available on modern devices - especially mobile keyboards. They should be able to select passwords from various languages and include pictograms. Prior to hashing the entropy of the user's entry should not be reduced, and password hashing libraries need to be able to use input that may contain a NULL byte.
```

**Review by szh — CHANGES_REQUESTED** [review]
I left some proofreading comments, but otherwise looks great! Thank you!

**Comment by kwwall** [comment]
@jmanico - Just an FYI, I am starting the requested code review today. Had been working on a PR for a new ESAPI release.

**Comment by kwwall** [comment]
@jmanico - Sorry I was late to this requested review. This is great work. The only minor comment that I would have is the paragraph under the "Peppering" section that states:
```text
For example, one peppering strategy is hashing the passwords as usual (using a password hashing algorithm) and then using HMAC or encrypting the hashes with a symmetrical encryption key before storing the password hash in the database, with the key acting as the pepper.
```
I personally would recommend changing the part "then using HMAC or encrypting the hashes with a symmetrical encryption key" (should be "symmetric encryption key", not "symmetrical" BTW), to instead ONLY refer to using HMACs. I'd recommend revising it as something like this:
```text
For example, one peppering strategy is hashing the passwords as usual (using a password hashing algorithm) and then using an  HMAC (e.g., HMAC-SHA256, HMAC-SHA512, depending on the desired output length) on the original password hash before storing the password hash in the database, with the pepper acting as the HMAC  key.
```
I recommend using an HMAC over symmetric encryption because there's just less to get wrong using HMACs than there is correctly implementing the encryption / decryption using a symmetric cipher.

Regardless of whether that change is made or not though, the last part of that paragraph, where it states:
```text
... with the key acting as the pepper.
```
is incorrect. It should say
```text
... with the pepper acting as the key.
```
Otherwise, looks good.

**Pull request merged**